Composer/packages/client/src/components/ProjectTree/index.tsx

```diff
@@ -107,17 +107,18 @@ export const ProjectTree: React.FC<IProjectTreeProps> = (props) => {
 
   const onRenderHeader = (props: IGroupHeaderProps) => {
     const toggleCollapse = (): void => {
-      groupRef.current!.toggleCollapseAll(true);
-      props.onToggleCollapse!(props.group!);
-      onSelect(props.group!.key);
+      if (props.group == null) return;
+      groupRef.current?.toggleCollapseAll(true);
+      props.onToggleCollapse?.(props.group);
+      onSelect(props.group.key);
     };
     return (
-      <span ref={props.group && props.group.data.isRoot && addMainDialogRef} role="grid">
+      <span ref={props.group?.data.isRoot && addMainDialogRef} role="grid">
         <TreeItem
           depth={0}
-          isActive={!props.group!.isCollapsed}
-          isSubItemActive={!!selected}
-          link={props.group!.data}
+          isActive={!props.group?.isCollapsed}
+          isSubItemActive={selected != null}
```

Comment:
Is use of null warranted here? Should these variable be assigned null in the first place. The decision could be part of our style guide decision as well. Like can we get through almost all scenarios with just using undefined

Reply:
The null check using != will check for either null or undefined (using !== is the one for specifically null values).

Reply:
I agree with @a-b-r-o-w-n . !selected would solve what !=null solves right. We can use either === {value} or !value.

Reply:
Also do we see any use of null in our codebase. Should variables even be instantiated to null?

Reply:
What is the type of `let foo;`? This breaks type safety. Variable declarations must be assigned a value in the typescript world.